### ISSUE TITLE
Unexpected errors are reported when type imported from metadata implements interface property in way that doesn't follow C# language rules

### ISSUE THREAD

**Issue by AlekseyTs** [dotnet/roslyn#33226]
```

        [Fact]
        public void ExplicitBase_135_Class_02()
        {
            var ilSource = @"
.class interface public abstract auto ansi I1
{
  .method public hidebysig newslot specialname abstract virtual 
          instance char  get_F1() cil managed
  {
  } // end of method I1::get_F1

  .method public hidebysig newslot specialname abstract virtual 
          instance void  set_F1(char 'value') cil managed
  {
  } // end of method I1::set_F1

  .property instance char F1()
  {
    .get instance char I1::get_F1()
    .set instance void I1::set_F1(char)
  } // end of property I1::F1
} // end of class I1

.class public auto ansi beforefieldinit A
       extends System.Object
{
  .method public hidebysig specialname rtspecialname 
          instance void  .ctor() cil managed
  {
    // Code size       8 (0x8)
    .maxstack  8
    IL_0000:  ldarg.0
    IL_0001:  call       instance void System.Object::.ctor()
    IL_0006:  nop
    IL_0007:  ret
  } // end of method A::.ctor
} // end of class A

.class public auto ansi beforefieldinit B
       extends A
{
  .method public hidebysig newslot specialname virtual final 
          instance char  get_F1() cil managed
  {
    // Code size       3 (0x3)
    .maxstack  8
    IL_0000:  ldc.i4.s   66
    IL_0002:  ret
  } // end of method B::I1.get_F1

  .method public hidebysig specialname rtspecialname 
          instance void  .ctor() cil managed
  {
    // Code size       8 (0x8)
    .maxstack  8
    IL_0000:  ldarg.0
    IL_0001:  call       instance void A::.ctor()
    IL_0006:  nop
    IL_0007:  ret
  } // end of method B::.ctor
} // end of class B

.class public auto ansi beforefieldinit C
       extends B
       implements I1
{
  .method public hidebysig newslot specialname virtual final 
          instance void  set_F1(char 'value') cil managed
  {
    // Code size       12 (0xc)
    .maxstack  8
    IL_0000:  ldstr      ""set C""
    IL_0005:  call       void [mscorlib]System.Console::WriteLine(string)
    IL_000a:  nop
    IL_000b:  ret
  } // end of method C::I1.set_F1

  .method public hidebysig specialname rtspecialname 
          instance void  .ctor() cil managed
  {
    // Code size       8 (0x8)
    .maxstack  8
    IL_0000:  ldarg.0
    IL_0001:  call       instance void B::.ctor()
    IL_0006:  nop
    IL_0007:  ret
  } // end of method C::.ctor
} // end of class C
";

            var source1 =
@"
class Test2 : C, I1
{
    static void Main()
    {
        I1 x = new Test2();

        System.Console.WriteLine(x.F1);
        x.F1 = 'x';
    }
}
";
            var compilation1 = CreateCompilationWithIL(source1, ilSource, options: TestOptions.DebugExe);
            compilation1.VerifyDiagnostics(
                // (2,18): error CS0470: Method 'C.set_F1(char)' cannot implement interface accessor 'I1.F1.set' for type 'Test2'. Use an explicit interface implementation.
                // class Test2 : C, I1
                Diagnostic(ErrorCode.ERR_MethodImplementingAccessor, "I1").WithArguments("C.set_F1(char)", "I1.F1.set", "Test2").WithLocation(2, 18),
                // (2,18): error CS0470: Method 'B.get_F1()' cannot implement interface accessor 'I1.F1.get' for type 'Test2'. Use an explicit interface implementation.
                // class Test2 : C, I1
                Diagnostic(ErrorCode.ERR_MethodImplementingAccessor, "I1").WithArguments("B.get_F1()", "I1.F1.get", "Test2").WithLocation(2, 18)
                );

            var i1 = compilation1.GetTypeByMetadataName("I1");
            var i1F1 = i1.GetMember<PropertySymbol>("F1");
            var test2 = compilation1.GetTypeByMetadataName("Test2");

            Assert.Null(test2.FindImplementationForInterfaceMember(i1F1));
            Assert.Equal("void C.set_F1(System.Char value)", test2.FindImplementationForInterfaceMember(i1F1.SetMethod).ToTestDisplayString());
            Assert.Equal("System.Char B.get_F1()", test2.FindImplementationForInterfaceMember(i1F1.GetMethod).ToTestDisplayString());
            Assert.Null(compilation1.GetTypeByMetadataName("B").FindImplementationForInterfaceMember(i1F1.GetMethod));

            var source2 =
@"
class Test2 : C
{
    static void Main()
    {
        I1 x = new Test2();

        System.Console.WriteLine(x.F1);
        x.F1 = 'x';
    }
}
";
            var compilation2 = CreateCompilationWithIL(source2, ilSource, options: TestOptions.DebugExe);
            compilation2.VerifyDiagnostics();

            i1 = compilation2.GetTypeByMetadataName("I1");
            i1F1 = i1.GetMember<PropertySymbol>("F1");
            var c = compilation2.GetTypeByMetadataName("C");

            Assert.Null(c.FindImplementationForInterfaceMember(i1F1));
            Assert.Equal("void C.set_F1(System.Char value)", c.FindImplementationForInterfaceMember(i1F1.SetMethod).ToTestDisplayString());
            Assert.Equal("System.Char B.get_F1()", c.FindImplementationForInterfaceMember(i1F1.GetMethod).ToTestDisplayString());
            Assert.Null(compilation2.GetTypeByMetadataName("B").FindImplementationForInterfaceMember(i1F1.GetMethod));

            CompileAndVerify(compilation2, expectedOutput: @"
B
set C
");
        }
```

Observed: Errors are reported for compilation1.
Expected: No errors, class ```C``` implements the interface 
```
.class public auto ansi beforefieldinit C
       extends B
       implements I1
``` 
Also, compilation2 succesfuly runs and verifies that runtime considers those methods implementing the interface.

Types defined in IL look roughly like this:
```
interface I1
{
    char F1 { get; set; }
}

class A 
{
}

class B : A
{
    public virtual sealed char F1 { get => 'B'; }
}

class C : B, I1
{
    public virtual sealed char I1.F1 { set => System.Console.WriteLine("set C"); }
}
```

**Comment by AlekseyTs** [comment]
Here is another example:
```
        [Fact]
        [WorkItem(33226, "https://github.com/dotnet/roslyn/issues/33226")]
        public void ExplicitBase_135_Class_03()
        {
            var ilSource = @"
.class interface public abstract auto ansi I1
{
  .method public hidebysig newslot specialname abstract virtual 
          instance char  get_F1() cil managed
  {
  } // end of method I1::get_F1

  .method public hidebysig newslot specialname abstract virtual 
          instance void  set_F1(char 'value') cil managed
  {
  } // end of method I1::set_F1

  .property instance char F1()
  {
    .get instance char I1::get_F1()
    .set instance void I1::set_F1(char)
  } // end of property I1::F1
} // end of class I1

.class public auto ansi beforefieldinit A
       extends System.Object
       implements I1
{
  .method private hidebysig newslot specialname virtual final 
          instance char  I1.get_F1() cil managed
  {
    .override I1::get_F1
    // Code size       3 (0x3)
    .maxstack  8
    IL_0000:  ldc.i4.s   65
    IL_0002:  ret
  } // end of method A::I1.get_F1

  .method private hidebysig newslot specialname virtual final 
          instance void  I1.set_F1(char 'value') cil managed
  {
    .override I1::set_F1
    // Code size       12 (0xc)
    .maxstack  8
    IL_0000:  ldstr      ""set A""
    IL_0005:  call       void [mscorlib]System.Console::WriteLine(string)
    IL_000a:  nop
    IL_000b:  ret
  } // end of method A::I1.set_F1

  .method public hidebysig specialname rtspecialname 
          instance void  .ctor() cil managed
  {
    // Code size       8 (0x8)
    .maxstack  8
    IL_0000:  ldarg.0
    IL_0001:  call       instance void System.Object::.ctor()
    IL_0006:  nop
    IL_0007:  ret
  } // end of method A::.ctor

  .property instance char I1.F1()
  {
    .get instance char A::I1.get_F1()
    .set instance void A::I1.set_F1(char)
  } // end of property A::I1.F1
} // end of class A

.class public auto ansi beforefieldinit B
       extends A
{
  .method public hidebysig newslot specialname virtual final 
          instance char  get_F1() cil managed
  {
    // Code size       3 (0x3)
    .maxstack  8
    IL_0000:  ldc.i4.s   66
    IL_0002:  ret
  } // end of method B::I1.get_F1

  .method public hidebysig specialname rtspecialname 
          instance void  .ctor() cil managed
  {
    // Code size       8 (0x8)
    .maxstack  8
    IL_0000:  ldarg.0
    IL_0001:  call       instance void A::.ctor()
    IL_0006:  nop
    IL_0007:  ret
  } // end of method B::.ctor
} // end of class B

.class public auto ansi beforefieldinit C
       extends B
       implements I1
{
  .method public hidebysig newslot specialname virtual final 
          instance void  set_F1(char 'value') cil managed
  {
    // Code size       12 (0xc)
    .maxstack  8
    IL_0000:  ldstr      ""set C""
    IL_0005:  call       void [mscorlib]System.Console::WriteLine(string)
    IL_000a:  nop
    IL_000b:  ret
  } // end of method C::I1.set_F1

  .method public hidebysig specialname rtspecialname 
          instance void  .ctor() cil managed
  {
    // Code size       8 (0x8)
    .maxstack  8
    IL_0000:  ldarg.0
    IL_0001:  call       instance void B::.ctor()
    IL_0006:  nop
    IL_0007:  ret
  } // end of method C::.ctor
} // end of class C
";

            var source1 =
@"
class Test2 : C, I1
{
    static void Main()
    {
        I1 x = new Test2();

        System.Console.WriteLine(x.F1);
        x.F1 = 'x';
    }
}
";
            var compilation1 = CreateCompilationWithIL(source1, ilSource, options: TestOptions.DebugExe);
            compilation1.VerifyDiagnostics(
                // (2,18): error CS0470: Method 'C.set_F1(char)' cannot implement interface accessor 'I1.F1.set' for type 'Test2'. Use an explicit interface implementation.
                // class Test2 : C, I1
                Diagnostic(ErrorCode.ERR_MethodImplementingAccessor, "I1").WithArguments("C.set_F1(char)", "I1.F1.set", "Test2").WithLocation(2, 18),
                // (2,18): error CS0470: Method 'B.get_F1()' cannot implement interface accessor 'I1.F1.get' for type 'Test2'. Use an explicit interface implementation.
                // class Test2 : C, I1
                Diagnostic(ErrorCode.ERR_MethodImplementingAccessor, "I1").WithArguments("B.get_F1()", "I1.F1.get", "Test2").WithLocation(2, 18)
                );

            var i1 = compilation1.GetTypeByMetadataName("I1");
            var i1F1 = i1.GetMember<PropertySymbol>("F1");
            var test2 = compilation1.GetTypeByMetadataName("Test2");

            Assert.Equal("System.Char A.I1.F1 { get; set; }", test2.FindImplementationForInterfaceMember(i1F1).ToTestDisplayString());
            Assert.Equal("void C.set_F1(System.Char value)", test2.FindImplementationForInterfaceMember(i1F1.SetMethod).ToTestDisplayString());
            Assert.Equal("System.Char B.get_F1()", test2.FindImplementationForInterfaceMember(i1F1.GetMethod).ToTestDisplayString());

            var source2 =
@"
class Test2 : C
{
    static void Main()
    {
        I1 x = new Test2();

        System.Console.WriteLine(x.F1);
        x.F1 = 'x';
    }
}
";
            var compilation2 = CreateCompilationWithIL(source2, ilSource, options: TestOptions.DebugExe);
            compilation2.VerifyDiagnostics();

            i1 = compilation2.GetTypeByMetadataName("I1");
            i1F1 = i1.GetMember<PropertySymbol>("F1");
            var c = compilation2.GetTypeByMetadataName("C");

            Assert.Equal("System.Char A.I1.F1 { get; set; }", c.FindImplementationForInterfaceMember(i1F1).ToTestDisplayString());
            Assert.Equal("void C.set_F1(System.Char value)", c.FindImplementationForInterfaceMember(i1F1.SetMethod).ToTestDisplayString());
            Assert.Equal("System.Char B.get_F1()", c.FindImplementationForInterfaceMember(i1F1.GetMethod).ToTestDisplayString());

            CompileAndVerify(compilation2, expectedOutput: @"
A
set C
");
        }
```

Observed:
- Errors are reported for compilation1.
- ```c.FindImplementationForInterfaceMember(i1F1)``` returns ```System.Char A.I1.F1 { get; set; }```
- ```c.FindImplementationForInterfaceMember(i1F1.GetMethod)``` returns ```B.get_F1```

Expected:
- No errors, class ```C``` implements the interface
- ```c.FindImplementationForInterfaceMember(i1F1)``` returns null because accessors of ```System.Char A.I1.F1 { get; set; }``` do not imoplement the interface for type ```C```.
- ```c.FindImplementationForInterfaceMember(i1F1.GetMethod)``` returns ```A.F1.get_F1```

```
.class public auto ansi beforefieldinit C
       extends B
       implements I1
```
Also, compilation2 succesfuly runs and verifies that runtime considers 'I1.F1.get' and 'I1.F1.set' methods implementing the interface.

Types defined in IL look roughly like this:

```
interface I1
{
    char F1 { get; set; }
}

class A : I1
{
    char I1.F1 { get => 'A'; set => System.Console.WriteLine("set A"); }
}

class B : A
{
    public virtual sealed char get_F1 => 'B';
}

class C : B, I1
{
    public virtual sealed void set_F1 (char value) => System.Console.WriteLine("set C");
}
```

**Comment by AlekseyTs** [comment]
More variations:
```
        [Fact]
        [WorkItem(33226, "https://github.com/dotnet/roslyn/issues/33226")]
        public void ExplicitBase_135_Class_04()
        {
            var ilSource = @"
.class interface public abstract auto ansi I1
{
  .method public hidebysig newslot specialname abstract virtual 
          instance char  get_F1() cil managed
  {
  } // end of method I1::get_F1

  .method public hidebysig newslot specialname abstract virtual 
          instance void  set_F1(char 'value') cil managed
  {
  } // end of method I1::set_F1

  .property instance char F1()
  {
    .get instance char I1::get_F1()
    .set instance void I1::set_F1(char)
  } // end of property I1::F1
} // end of class I1

.class public auto ansi beforefieldinit A
       extends System.Object
       implements I1
{
  .method public hidebysig newslot specialname virtual final 
          instance char  get_F1() cil managed
  {
    // Code size       3 (0x3)
    .maxstack  8
    IL_0000:  ldc.i4.s   65
    IL_0002:  ret
  } // end of method A::I1.get_F1

  .method public hidebysig newslot specialname virtual final 
          instance void  set_F1(char 'value') cil managed
  {
    // Code size       12 (0xc)
    .maxstack  8
    IL_0000:  ldstr      ""set A""
    IL_0005:  call       void [mscorlib]System.Console::WriteLine(string)
    IL_000a:  nop
    IL_000b:  ret
  } // end of method A::I1.set_F1

  .method public hidebysig specialname rtspecialname 
          instance void  .ctor() cil managed
  {
    // Code size       8 (0x8)
    .maxstack  8
    IL_0000:  ldarg.0
    IL_0001:  call       instance void System.Object::.ctor()
    IL_0006:  nop
    IL_0007:  ret
  } // end of method A::.ctor

  .property instance char F1()
  {
    .get instance char A::get_F1()
    .set instance void A::set_F1(char)
  } // end of property A::I1.F1
} // end of class A

.class public auto ansi beforefieldinit B
       extends A
{
  .method public hidebysig newslot specialname virtual final 
          instance char  get_F1() cil managed
  {
    // Code size       3 (0x3)
    .maxstack  8
    IL_0000:  ldc.i4.s   66
    IL_0002:  ret
  } // end of method B::I1.get_F1

  .method public hidebysig specialname rtspecialname 
          instance void  .ctor() cil managed
  {
    // Code size       8 (0x8)
    .maxstack  8
    IL_0000:  ldarg.0
    IL_0001:  call       instance void A::.ctor()
    IL_0006:  nop
    IL_0007:  ret
  } // end of method B::.ctor
} // end of class B

.class public auto ansi beforefieldinit C
       extends B
       implements I1
{
  .method public hidebysig newslot specialname virtual final 
          instance void  set_F1(char 'value') cil managed
  {
    // Code size       12 (0xc)
    .maxstack  8
    IL_0000:  ldstr      ""set C""
    IL_0005:  call       void [mscorlib]System.Console::WriteLine(string)
    IL_000a:  nop
    IL_000b:  ret
  } // end of method C::I1.set_F1

  .method public hidebysig specialname rtspecialname 
          instance void  .ctor() cil managed
  {
    // Code size       8 (0x8)
    .maxstack  8
    IL_0000:  ldarg.0
    IL_0001:  call       instance void B::.ctor()
    IL_0006:  nop
    IL_0007:  ret
  } // end of method C::.ctor
} // end of class C
";

            var source1 =
@"
class Test2 : C, I1
{
    static void Main()
    {
        I1 x = new Test2();

        System.Console.WriteLine(x.F1);
        x.F1 = 'x';
    }
}
";
            var compilation1 = CreateCompilationWithIL(source1, ilSource, options: TestOptions.DebugExe);
            compilation1.VerifyDiagnostics(
                // (2,18): error CS0470: Method 'C.set_F1(char)' cannot implement interface accessor 'I1.F1.set' for type 'Test2'. Use an explicit interface implementation.
                // class Test2 : C, I1
                Diagnostic(ErrorCode.ERR_MethodImplementingAccessor, "I1").WithArguments("C.set_F1(char)", "I1.F1.set", "Test2").WithLocation(2, 18),
                // (2,18): error CS0470: Method 'B.get_F1()' cannot implement interface accessor 'I1.F1.get' for type 'Test2'. Use an explicit interface implementation.
                // class Test2 : C, I1
                Diagnostic(ErrorCode.ERR_MethodImplementingAccessor, "I1").WithArguments("B.get_F1()", "I1.F1.get", "Test2").WithLocation(2, 18)
                );

            var i1 = compilation1.GetTypeByMetadataName("I1");
            var i1F1 = i1.GetMember<PropertySymbol>("F1");
            var test2 = compilation1.GetTypeByMetadataName("Test2");

            Assert.Equal("System.Char A.F1 { get; set; }", test2.FindImplementationForInterfaceMember(i1F1).ToTestDisplayString());
            Assert.Equal("void C.set_F1(System.Char value)", test2.FindImplementationForInterfaceMember(i1F1.SetMethod).ToTestDisplayString());
            Assert.Equal("System.Char B.get_F1()", test2.FindImplementationForInterfaceMember(i1F1.GetMethod).ToTestDisplayString());

            var source2 =
@"
class Test2 : C
{
    static void Main()
    {
        I1 x = new Test2();

        System.Console.WriteLine(x.F1);
        x.F1 = 'x';
    }
}
";
            var compilation2 = CreateCompilationWithIL(source2, ilSource, options: TestOptions.DebugExe);
            compilation2.VerifyDiagnostics();

            i1 = compilation2.GetTypeByMetadataName("I1");
            i1F1 = i1.GetMember<PropertySymbol>("F1");
            var c = compilation2.GetTypeByMetadataName("C");

            Assert.Equal("System.Char A.F1 { get; set; }", c.FindImplementationForInterfaceMember(i1F1).ToTestDisplayString());
            Assert.Equal("void C.set_F1(System.Char value)", c.FindImplementationForInterfaceMember(i1F1.SetMethod).ToTestDisplayString());
            Assert.Equal("System.Char B.get_F1()", c.FindImplementationForInterfaceMember(i1F1.GetMethod).ToTestDisplayString());

            CompileAndVerify(compilation2, expectedOutput: @"
A
set C
");
        }

        [Fact]
        [WorkItem(33226, "https://github.com/dotnet/roslyn/issues/33226")]
        public void ExplicitBase_134_Class_01()
        {
            var ilSource = @"
.class interface public abstract auto ansi I1
{
  .method public hidebysig newslot specialname abstract virtual 
          instance char  get_F1() cil managed
  {
  } // end of method I1::get_F1

  .method public hidebysig newslot specialname abstract virtual 
          instance void  set_F1(char 'value') cil managed
  {
  } // end of method I1::set_F1

  .property instance char F1()
  {
    .get instance char I1::get_F1()
    .set instance void I1::set_F1(char)
  } // end of property I1::F1
} // end of class I1

.class public auto ansi beforefieldinit A
       extends System.Object
       implements I1
{
  .method private hidebysig newslot specialname virtual final 
          instance char  I1.get_F1() cil managed
  {
    .override I1::get_F1
    // Code size       3 (0x3)
    .maxstack  8
    IL_0000:  ldc.i4.s   65
    IL_0002:  ret
  } // end of method A::I1.get_F1

  .method private hidebysig newslot specialname virtual final 
          instance void  I1.set_F1(char 'value') cil managed
  {
    .override I1::set_F1
    // Code size       12 (0xc)
    .maxstack  8
    IL_0000:  ldstr      ""set A""
    IL_0005:  call       void [mscorlib]System.Console::WriteLine(string)
    IL_000a:  nop
    IL_000b:  ret
  } // end of method A::I1.set_F1

  .method public hidebysig specialname rtspecialname 
          instance void  .ctor() cil managed
  {
    // Code size       8 (0x8)
    .maxstack  8
    IL_0000:  ldarg.0
    IL_0001:  call       instance void System.Object::.ctor()
    IL_0006:  nop
    IL_0007:  ret
  } // end of method A::.ctor

  .property instance char I1.F1()
  {
    .get instance char A::I1.get_F1()
    .set instance void A::I1.set_F1(char)
  } // end of property A::I1.F1
} // end of class A

.class public auto ansi beforefieldinit B
       extends A
       implements I1
{
  .method private hidebysig newslot specialname virtual final 
          instance char  I1.get_F1() cil managed
  {
    .override I1::get_F1
    // Code size       3 (0x3)
    .maxstack  8
    IL_0000:  ldc.i4.s   66
    IL_0002:  ret
  } // end of method B::I1.get_F1

  .method public hidebysig specialname rtspecialname 
          instance void  .ctor() cil managed
  {
    // Code size       8 (0x8)
    .maxstack  8
    IL_0000:  ldarg.0
    IL_0001:  call       instance void A::.ctor()
    IL_0006:  nop
    IL_0007:  ret
  } // end of method B::.ctor

  .property instance char I1.F1()
  {
    .get instance char B::I1.get_F1()
  } // end of property B::I1.F1
} // end of class B

.class public auto ansi beforefieldinit C
       extends B
       implements I1
{
  .method private hidebysig newslot specialname virtual final 
          instance void  I1.set_F1(char 'value') cil managed
  {
    .override I1::set_F1
    // Code size       12 (0xc)
    .maxstack  8
    IL_0000:  ldstr      ""set C""
    IL_0005:  call       void [mscorlib]System.Console::WriteLine(string)
    IL_000a:  nop
    IL_000b:  ret
  } // end of method C::I1.set_F1

  .method public hidebysig specialname rtspecialname 
          instance void  .ctor() cil managed
  {
    // Code size       8 (0x8)
    .maxstack  8
    IL_0000:  ldarg.0
    IL_0001:  call       instance void B::.ctor()
    IL_0006:  nop
    IL_0007:  ret
  } // end of method C::.ctor

  .property instance char I1.F1()
  {
    .set instance void C::I1.set_F1(char)
  } // end of property C::I1.F1
} // end of class C
";

            var source1 =
@"
class Test2 : C, I1
{
    static void Main()
    {
        I1 x = new Test2();

        System.Console.WriteLine(x.F1);
        x.F1 = 'x';
    }
}
";
            var compilation1 = CreateCompilationWithIL(source1, ilSource, options: TestOptions.DebugExe);
            compilation1.VerifyDiagnostics();

            var i1 = compilation1.GetTypeByMetadataName("I1");
            var i1F1 = i1.GetMember<PropertySymbol>("F1");
            var test2 = compilation1.GetTypeByMetadataName("Test2");

            Assert.Equal("System.Char C.F1 { set; }", test2.FindImplementationForInterfaceMember(i1F1).ToTestDisplayString());
            Assert.Equal("void C.F1.set", test2.FindImplementationForInterfaceMember(i1F1.SetMethod).ToTestDisplayString());
            Assert.Equal("System.Char B.F1.get", test2.FindImplementationForInterfaceMember(i1F1.GetMethod).ToTestDisplayString());

            CompileAndVerify(compilation1, expectedOutput: @"
B
set C
");

            var source2 =
@"
class Test2 : C
{
    static void Main()
    {
        I1 x = new Test2();

        System.Console.WriteLine(x.F1);
        x.F1 = 'x';
    }
}
";
            var compilation2 = CreateCompilationWithIL(source2, ilSource, options: TestOptions.DebugExe);
            compilation2.VerifyDiagnostics();

            i1 = compilation2.GetTypeByMetadataName("I1");
            i1F1 = i1.GetMember<PropertySymbol>("F1");
            var c = compilation2.GetTypeByMetadataName("C");

            Assert.Equal("System.Char C.F1 { set; }", c.FindImplementationForInterfaceMember(i1F1).ToTestDisplayString());
            Assert.Equal("void C.F1.set", c.FindImplementationForInterfaceMember(i1F1.SetMethod).ToTestDisplayString());
            Assert.Equal("System.Char B.F1.get", c.FindImplementationForInterfaceMember(i1F1.GetMethod).ToTestDisplayString());

            CompileAndVerify(compilation2, expectedOutput: @"
B
set C
");
        }

        [Fact]
        [WorkItem(33226, "https://github.com/dotnet/roslyn/issues/33226")]
        public void ExplicitBase_134_Class_02()
        {
            var ilSource = @"
.class interface public abstract auto ansi I1
{
  .method public hidebysig newslot specialname abstract virtual 
          instance char  get_F1() cil managed
  {
  } // end of method I1::get_F1

  .method public hidebysig newslot specialname abstract virtual 
          instance void  set_F1(char 'value') cil managed
  {
  } // end of method I1::set_F1

  .property instance char F1()
  {
    .get instance char I1::get_F1()
    .set instance void I1::set_F1(char)
  } // end of property I1::F1
} // end of class I1

.class public auto ansi beforefieldinit A
       extends System.Object
{
  .method public hidebysig specialname rtspecialname 
          instance void  .ctor() cil managed
  {
    // Code size       8 (0x8)
    .maxstack  8
    IL_0000:  ldarg.0
    IL_0001:  call       instance void System.Object::.ctor()
    IL_0006:  nop
    IL_0007:  ret
  } // end of method A::.ctor
} // end of class A

.class public auto ansi beforefieldinit B
       extends A
{
  .method public hidebysig newslot specialname virtual final 
          instance char  get_F1() cil managed
  {
    // Code size       3 (0x3)
    .maxstack  8
    IL_0000:  ldc.i4.s   66
    IL_0002:  ret
  } // end of method B::I1.get_F1

  .method public hidebysig specialname rtspecialname 
          instance void  .ctor() cil managed
  {
    // Code size       8 (0x8)
    .maxstack  8
    IL_0000:  ldarg.0
    IL_0001:  call       instance void A::.ctor()
    IL_0006:  nop
    IL_0007:  ret
  } // end of method B::.ctor

  .property instance char F1()
  {
    .get instance char B::get_F1()
  } // end of property B::I1.F1
} // end of class B

.class public auto ansi beforefieldinit C
       extends B
       implements I1
{
  .method public hidebysig newslot specialname virtual final 
          instance void  set_F1(char 'value') cil managed
  {
    // Code size       12 (0xc)
    .maxstack  8
    IL_0000:  ldstr      ""set C""
    IL_0005:  call       void [mscorlib]System.Console::WriteLine(string)
    IL_000a:  nop
    IL_000b:  ret
  } // end of method C::I1.set_F1

  .method public hidebysig specialname rtspecialname 
          instance void  .ctor() cil managed
  {
    // Code size       8 (0x8)
    .maxstack  8
    IL_0000:  ldarg.0
    IL_0001:  call       instance void B::.ctor()
    IL_0006:  nop
    IL_0007:  ret
  } // end of method C::.ctor

  .property instance char F1()
  {
    .set instance void C::set_F1(char)
  } // end of property C::I1.F1
} // end of class C
";

            var source =
@"
class Test2 : C, I1
{
    static void Main()
    {
        I1 x = new Test2();

        System.Console.WriteLine(x.F1);
        x.F1 = 'x';
    }
}
";
            var compilation1 = CreateCompilationWithIL(source, ilSource, options: TestOptions.DebugExe);
            compilation1.VerifyDiagnostics();

            var i1 = compilation1.GetTypeByMetadataName("I1");
            var i1F1 = i1.GetMember<PropertySymbol>("F1");
            var test2 = compilation1.GetTypeByMetadataName("Test2");

            Assert.Equal("System.Char C.F1 { set; }", test2.FindImplementationForInterfaceMember(i1F1).ToTestDisplayString());
            Assert.Equal("void C.F1.set", test2.FindImplementationForInterfaceMember(i1F1.SetMethod).ToTestDisplayString());
            Assert.Null(test2.FindImplementationForInterfaceMember(i1F1.GetMethod));
            Assert.Null(compilation1.GetTypeByMetadataName("B").FindImplementationForInterfaceMember(i1F1.GetMethod));

            CompileAndVerify(compilation1, expectedOutput: @"
B
set C
");

            var source2 =
@"
class Test2 : C
{
    static void Main()
    {
        I1 x = new Test2();

        System.Console.WriteLine(x.F1);
        x.F1 = 'x';
    }
}
";
            var compilation2 = CreateCompilationWithIL(source2, ilSource, options: TestOptions.DebugExe);
            compilation2.VerifyDiagnostics();

            i1 = compilation2.GetTypeByMetadataName("I1");
            i1F1 = i1.GetMember<PropertySymbol>("F1");
            var c = compilation2.GetTypeByMetadataName("C");

            Assert.Equal("System.Char C.F1 { set; }", c.FindImplementationForInterfaceMember(i1F1).ToTestDisplayString());
            Assert.Equal("void C.F1.set", c.FindImplementationForInterfaceMember(i1F1.SetMethod).ToTestDisplayString());
            Assert.Null(c.FindImplementationForInterfaceMember(i1F1.GetMethod));

            CompileAndVerify(compilation2, expectedOutput: @"
B
set C
");
        }

        [Fact]
        [WorkItem(33226, "https://github.com/dotnet/roslyn/issues/33226")]
        public void ExplicitBase_134_Class_03()
        {
            var ilSource = @"
.class interface public abstract auto ansi I1
{
  .method public hidebysig newslot specialname abstract virtual 
          instance char  get_F1() cil managed
  {
  } // end of method I1::get_F1

  .method public hidebysig newslot specialname abstract virtual 
          instance void  set_F1(char 'value') cil managed
  {
  } // end of method I1::set_F1

  .property instance char F1()
  {
    .get instance char I1::get_F1()
    .set instance void I1::set_F1(char)
  } // end of property I1::F1
} // end of class I1

.class public auto ansi beforefieldinit A
       extends System.Object
       implements I1
{
  .method private hidebysig newslot specialname virtual final 
          instance char  I1.get_F1() cil managed
  {
    .override I1::get_F1
    // Code size       3 (0x3)
    .maxstack  8
    IL_0000:  ldc.i4.s   65
    IL_0002:  ret
  } // end of method A::I1.get_F1

  .method private hidebysig newslot specialname virtual final 
          instance void  I1.set_F1(char 'value') cil managed
  {
    .override I1::set_F1
    // Code size       12 (0xc)
    .maxstack  8
    IL_0000:  ldstr      ""set A""
    IL_0005:  call       void [mscorlib]System.Console::WriteLine(string)
    IL_000a:  nop
    IL_000b:  ret
  } // end of method A::I1.set_F1

  .method public hidebysig specialname rtspecialname 
          instance void  .ctor() cil managed
  {
    // Code size       8 (0x8)
    .maxstack  8
    IL_0000:  ldarg.0
    IL_0001:  call       instance void System.Object::.ctor()
    IL_0006:  nop
    IL_0007:  ret
  } // end of method A::.ctor

  .property instance char I1.F1()
  {
    .get instance char A::I1.get_F1()
    .set instance void A::I1.set_F1(char)
  } // end of property A::I1.F1
} // end of class A

.class public auto ansi beforefieldinit B
       extends A
{
  .method public hidebysig newslot specialname virtual final 
          instance char  get_F1() cil managed
  {
    // Code size       3 (0x3)
    .maxstack  8
    IL_0000:  ldc.i4.s   66
    IL_0002:  ret
  } // end of method B::I1.get_F1

  .method public hidebysig specialname rtspecialname 
          instance void  .ctor() cil managed
  {
    // Code size       8 (0x8)
    .maxstack  8
    IL_0000:  ldarg.0
    IL_0001:  call       instance void A::.ctor()
    IL_0006:  nop
    IL_0007:  ret
  } // end of method B::.ctor

  .property instance char F1()
  {
    .get instance char B::get_F1()
  } // end of property B::I1.F1
} // end of class B

.class public auto ansi beforefieldinit C
       extends B
       implements I1
{
  .method public hidebysig newslot specialname virtual final 
          instance void  set_F1(char 'value') cil managed
  {
    // Code size       12 (0xc)
    .maxstack  8
    IL_0000:  ldstr      ""set C""
    IL_0005:  call       void [mscorlib]System.Console::WriteLine(string)
    IL_000a:  nop
    IL_000b:  ret
  } // end of method C::I1.set_F1

  .method public hidebysig specialname rtspecialname 
          instance void  .ctor() cil managed
  {
    // Code size       8 (0x8)
    .maxstack  8
    IL_0000:  ldarg.0
    IL_0001:  call       instance void B::.ctor()
    IL_0006:  nop
    IL_0007:  ret
  } // end of method C::.ctor

  .property instance char F1()
  {
    .set instance void C::set_F1(char)
  } // end of property C::I1.F1
} // end of class C
";

            var source =
@"
class Test2 : C, I1
{
    static void Main()
    {
        I1 x = new Test2();

        System.Console.WriteLine(x.F1);
        x.F1 = 'x';
    }
}
";
            var compilation1 = CreateCompilationWithIL(source, ilSource, options: TestOptions.DebugExe);
            compilation1.VerifyDiagnostics();

            var i1 = compilation1.GetTypeByMetadataName("I1");
            var i1F1 = i1.GetMember<PropertySymbol>("F1");
            var test2 = compilation1.GetTypeByMetadataName("Test2");

            Assert.Equal("System.Char C.F1 { set; }", test2.FindImplementationForInterfaceMember(i1F1).ToTestDisplayString());
            Assert.Equal("void C.F1.set", test2.FindImplementationForInterfaceMember(i1F1.SetMethod).ToTestDisplayString());
            Assert.Null(test2.FindImplementationForInterfaceMember(i1F1.GetMethod));

            CompileAndVerify(compilation1, expectedOutput: @"
A
set C
");

            var source2 =
@"
class Test2 : C
{
    static void Main()
    {
        I1 x = new Test2();

        System.Console.WriteLine(x.F1);
        x.F1 = 'x';
    }
}
";
            var compilation2 = CreateCompilationWithIL(source2, ilSource, options: TestOptions.DebugExe);
            compilation2.VerifyDiagnostics();

            i1 = compilation2.GetTypeByMetadataName("I1");
            i1F1 = i1.GetMember<PropertySymbol>("F1");
            var c = compilation2.GetTypeByMetadataName("C");

            Assert.Equal("System.Char C.F1 { set; }", c.FindImplementationForInterfaceMember(i1F1).ToTestDisplayString());
            Assert.Equal("void C.F1.set", c.FindImplementationForInterfaceMember(i1F1.SetMethod).ToTestDisplayString());
            Assert.Null(c.FindImplementationForInterfaceMember(i1F1.GetMethod));

            CompileAndVerify(compilation2, expectedOutput: @"
A
set C
");
        }

        [Fact]
        [WorkItem(33226, "https://github.com/dotnet/roslyn/issues/33226")]
        public void ExplicitBase_134_Class_04()
        {
            var ilSource = @"
.class interface public abstract auto ansi I1
{
  .method public hidebysig newslot specialname abstract virtual 
          instance char  get_F1() cil managed
  {
  } // end of method I1::get_F1

  .method public hidebysig newslot specialname abstract virtual 
          instance void  set_F1(char 'value') cil managed
  {
  } // end of method I1::set_F1

  .property instance char F1()
  {
    .get instance char I1::get_F1()
    .set instance void I1::set_F1(char)
  } // end of property I1::F1
} // end of class I1

.class public auto ansi beforefieldinit A
       extends System.Object
       implements I1
{
  .method public hidebysig newslot specialname virtual final 
          instance char  get_F1() cil managed
  {
    // Code size       3 (0x3)
    .maxstack  8
    IL_0000:  ldc.i4.s   65
    IL_0002:  ret
  } // end of method A::I1.get_F1

  .method public hidebysig newslot specialname virtual final 
          instance void  set_F1(char 'value') cil managed
  {
    // Code size       12 (0xc)
    .maxstack  8
    IL_0000:  ldstr      ""set A""
    IL_0005:  call       void [mscorlib]System.Console::WriteLine(string)
    IL_000a:  nop
    IL_000b:  ret
  } // end of method A::I1.set_F1

  .method public hidebysig specialname rtspecialname 
          instance void  .ctor() cil managed
  {
    // Code size       8 (0x8)
    .maxstack  8
    IL_0000:  ldarg.0
    IL_0001:  call       instance void System.Object::.ctor()
    IL_0006:  nop
    IL_0007:  ret
  } // end of method A::.ctor

  .property instance char F1()
  {
    .get instance char A::get_F1()
    .set instance void A::set_F1(char)
  } // end of property A::I1.F1
} // end of class A

.class public auto ansi beforefieldinit B
       extends A
{
  .method public hidebysig newslot specialname virtual final 
          instance char  get_F1() cil managed
  {
    // Code size       3 (0x3)
    .maxstack  8
    IL_0000:  ldc.i4.s   66
    IL_0002:  ret
  } // end of method B::I1.get_F1

  .method public hidebysig specialname rtspecialname 
          instance void  .ctor() cil managed
  {
    // Code size       8 (0x8)
    .maxstack  8
    IL_0000:  ldarg.0
    IL_0001:  call       instance void A::.ctor()
    IL_0006:  nop
    IL_0007:  ret
  } // end of method B::.ctor

  .property instance char F1()
  {
    .get instance char B::get_F1()
  } // end of property B::I1.F1
} // end of class B

.class public auto ansi beforefieldinit C
       extends B
       implements I1
{
  .method public hidebysig newslot specialname virtual final 
          instance void  set_F1(char 'value') cil managed
  {
    // Code size       12 (0xc)
    .maxstack  8
    IL_0000:  ldstr      ""set C""
    IL_0005:  call       void [mscorlib]System.Console::WriteLine(string)
    IL_000a:  nop
    IL_000b:  ret
  } // end of method C::I1.set_F1

  .method public hidebysig specialname rtspecialname 
          instance void  .ctor() cil managed
  {
    // Code size       8 (0x8)
    .maxstack  8
    IL_0000:  ldarg.0
    IL_0001:  call       instance void B::.ctor()
    IL_0006:  nop
    IL_0007:  ret
  } // end of method C::.ctor

  .property instance char F1()
  {
    .set instance void C::set_F1(char)
  } // end of property C::I1.F1
} // end of class C
";

            var source =
@"
class Test2 : C, I1
{
    static void Main()
    {
        I1 x = new Test2();

        System.Console.WriteLine(x.F1);
        x.F1 = 'x';
    }
}
";
            var compilation1 = CreateCompilationWithIL(source, ilSource, options: TestOptions.DebugExe);
            compilation1.VerifyDiagnostics();

            var i1 = compilation1.GetTypeByMetadataName("I1");
            var i1F1 = i1.GetMember<PropertySymbol>("F1");
            var test2 = compilation1.GetTypeByMetadataName("Test2");

            Assert.Equal("System.Char C.F1 { set; }", test2.FindImplementationForInterfaceMember(i1F1).ToTestDisplayString());
            Assert.Equal("void C.F1.set", test2.FindImplementationForInterfaceMember(i1F1.SetMethod).ToTestDisplayString());
            Assert.Null(test2.FindImplementationForInterfaceMember(i1F1.GetMethod));

            CompileAndVerify(compilation1, expectedOutput: @"
A
set C
");

            var source2 =
@"
class Test2 : C
{
    static void Main()
    {
        I1 x = new Test2();

        System.Console.WriteLine(x.F1);
        x.F1 = 'x';
    }
}
";
            var compilation2 = CreateCompilationWithIL(source2, ilSource, options: TestOptions.DebugExe);
            compilation2.VerifyDiagnostics();

            i1 = compilation2.GetTypeByMetadataName("I1");
            i1F1 = i1.GetMember<PropertySymbol>("F1");
            var c = compilation2.GetTypeByMetadataName("C");

            Assert.Equal("System.Char C.F1 { set; }", c.FindImplementationForInterfaceMember(i1F1).ToTestDisplayString());
            Assert.Equal("void C.F1.set", c.FindImplementationForInterfaceMember(i1F1.SetMethod).ToTestDisplayString());
            Assert.Null(c.FindImplementationForInterfaceMember(i1F1.GetMethod));

            CompileAndVerify(compilation2, expectedOutput: @"
A
set C
");
        }

        [Fact]
        [WorkItem(33226, "https://github.com/dotnet/roslyn/issues/33226")]
        public void ExplicitBase_134_Class_05()
        {
            var ilSource = @"
.class interface public abstract auto ansi I1
{
  .method public hidebysig newslot specialname abstract virtual 
          instance char  get_F1() cil managed
  {
  } // end of method I1::get_F1

  .method public hidebysig newslot specialname abstract virtual 
          instance void  set_F1(char 'value') cil managed
  {
  } // end of method I1::set_F1

  .property instance char F1()
  {
    .get instance char I1::get_F1()
    .set instance void I1::set_F1(char)
  } // end of property I1::F1
} // end of class I1

.class public auto ansi beforefieldinit A
       extends System.Object
       implements I1
{
  .method public hidebysig newslot specialname virtual final 
          instance char  get_F1() cil managed
  {
    // Code size       3 (0x3)
    .maxstack  8
    IL_0000:  ldc.i4.s   65
    IL_0002:  ret
  } // end of method A::I1.get_F1

  .method public hidebysig newslot specialname virtual final 
          instance void  set_F1(char 'value') cil managed
  {
    // Code size       12 (0xc)
    .maxstack  8
    IL_0000:  ldstr      ""set A""
    IL_0005:  call       void [mscorlib]System.Console::WriteLine(string)
    IL_000a:  nop
    IL_000b:  ret
  } // end of method A::I1.set_F1

  .method public hidebysig specialname rtspecialname 
          instance void  .ctor() cil managed
  {
    // Code size       8 (0x8)
    .maxstack  8
    IL_0000:  ldarg.0
    IL_0001:  call       instance void System.Object::.ctor()
    IL_0006:  nop
    IL_0007:  ret
  } // end of method A::.ctor

  .property instance char F1()
  {
    .get instance char A::get_F1()
    .set instance void A::set_F1(char)
  } // end of property A::I1.F1
} // end of class A

.class public auto ansi beforefieldinit B
       extends A
       implements I1
{
  .method private hidebysig newslot specialname virtual final 
          instance char  I1.get_F1() cil managed
  {
    .override I1::get_F1
    // Code size       3 (0x3)
    .maxstack  8
    IL_0000:  ldc.i4.s   66
    IL_0002:  ret
  } // end of method B::I1.get_F1

  .method public hidebysig specialname rtspecialname 
          instance void  .ctor() cil managed
  {
    // Code size       8 (0x8)
    .maxstack  8
    IL_0000:  ldarg.0
    IL_0001:  call       instance void A::.ctor()
    IL_0006:  nop
    IL_0007:  ret
  } // end of method B::.ctor

  .property instance char I1.F1()
  {
    .get instance char B::I1.get_F1()
  } // end of property B::I1.F1
} // end of class B

.class public auto ansi beforefieldinit C
       extends B
       implements I1
{
  .method private hidebysig newslot specialname virtual final 
          instance void  I1.set_F1(char 'value') cil managed
  {
    .override I1::set_F1
    // Code size       12 (0xc)
    .maxstack  8
    IL_0000:  ldstr      ""set C""
    IL_0005:  call       void [mscorlib]System.Console::WriteLine(string)
    IL_000a:  nop
    IL_000b:  ret
  } // end of method C::I1.set_F1

  .method public hidebysig specialname rtspecialname 
          instance void  .ctor() cil managed
  {
    // Code size       8 (0x8)
    .maxstack  8
    IL_0000:  ldarg.0
    IL_0001:  call       instance void B::.ctor()
    IL_0006:  nop
    IL_0007:  ret
  } // end of method C::.ctor

  .property instance char I1.F1()
  {
    .set instance void C::I1.set_F1(char)
  } // end of property C::I1.F1
} // end of class C
";

            var source =
@"
class Test2 : C, I1
{
    static void Main()
    {
        I1 x = new Test2();

        System.Console.WriteLine(x.F1);
        x.F1 = 'x';
    }
}
";
            var compilation1 = CreateCompilationWithIL(source, ilSource, options: TestOptions.DebugExe);
            compilation1.VerifyDiagnostics();

            var i1 = compilation1.GetTypeByMetadataName("I1");
            var i1F1 = i1.GetMember<PropertySymbol>("F1");
            var test2 = compilation1.GetTypeByMetadataName("Test2");

            Assert.Equal("System.Char C.F1 { set; }", test2.FindImplementationForInterfaceMember(i1F1).ToTestDisplayString());
            Assert.Equal("void C.F1.set", test2.FindImplementationForInterfaceMember(i1F1.SetMethod).ToTestDisplayString());
            Assert.Equal("System.Char B.F1.get", test2.FindImplementationForInterfaceMember(i1F1.GetMethod).ToTestDisplayString());

            CompileAndVerify(compilation1, expectedOutput: @"
B
set C
");

            var source2 =
@"
class Test2 : C
{
    static void Main()
    {
        I1 x = new Test2();

        System.Console.WriteLine(x.F1);
        x.F1 = 'x';
    }
}
";
            var compilation2 = CreateCompilationWithIL(source2, ilSource, options: TestOptions.DebugExe);
            compilation2.VerifyDiagnostics();

            i1 = compilation2.GetTypeByMetadataName("I1");
            i1F1 = i1.GetMember<PropertySymbol>("F1");
            var c = compilation2.GetTypeByMetadataName("C");

            Assert.Equal("System.Char C.F1 { set; }", c.FindImplementationForInterfaceMember(i1F1).ToTestDisplayString());
            Assert.Equal("void C.F1.set", c.FindImplementationForInterfaceMember(i1F1.SetMethod).ToTestDisplayString());
            Assert.Equal("System.Char B.F1.get", c.FindImplementationForInterfaceMember(i1F1.GetMethod).ToTestDisplayString());

            CompileAndVerify(compilation2, expectedOutput: @"
B
set C
");
        }
```
